app/src/main/java/org/metabrainz/android/presentation/features/search/SearchActivity.kt

```diff
@@ -210,5 +210,16 @@ class SearchActivity : AppCompatActivity(), SearchView.OnQueryTextListener {
         }
         return false
     }
+
+    override fun onBackPressed() {
+        if (!binding.gridView.isVisible){
```

Comment:
After considering the user experience, I found that when successful search results are shown to a user, the back press would have them show the screen to add artists, etc., hence introducing an extra back press for someone who's looking to exit the search screen. It's better to have the back press overridden for the case only when the search fails.

Reply:
Sir I changed the condition from ```!binding.gridView.isVisible``` to ```binding.noResult.isVisible```.
Is it fine or do I need to make other changes ??